platform/src/can.cpp

```diff
@@ -0,0 +1,318 @@
+#include "can.hpp"
+#include "string.h"
+
+static uint8_t messageMarkerGenerator = 0;
+static constexpr uint8_t dlc_to_data_length[16] = {
+    0,
+    1,
+    2,
+    3,
+    4,
+    5,
+    6,
+    7,
+    8,
+    12,
+    16,
+    20,
+    24,
+    32,
+    48,
+    64
+};
+
+CanMessageFilter::CanMessageFilter()
+    : filter{
+        0,
+        0,
+        0,
+        0,
+        0,
+        0,
+    } {}
+
+CanMessageFilter::CanMessageFilter(uint32_t filter_type,
+                                   uint32_t filter_config,
+                                   uint32_t filter_id1,
+                                   uint32_t filter_id2)
+    : filter{
+        FDCAN_STANDARD_ID,
+        0, //< Late INIT (Filter Index)
+        filter_type,
+        filter_config,
+        filter_id1,
+        filter_id2,
+    } {}
+
+CanMessage::CanMessage(CanMessageId id,
+                            uint8_t *data,
+                            uint32_t data_length)
+    : identifier(id),
+      message_marker(messageMarkerGenerator++),
+      data(data),
+      data_length(data_length) {}
+
+void CanMessage::set_id(uint32_t id) {
+    using Id = CanMessageId;
+    switch (id) {
```

Comment:
Is there a reason we can't take in the `id` as a `CanMessageId` instead of a `uint32_t`, so we can just directly assign to `identifier`?

Reply:
My argument would be type safety. Technically, we could get any message depending on the way that we configure filters; specifically, they could let in a message from the roboteque that we weren't expecting and didn't account for when building filters. In this case, assigning an `uint32_t` to the enum would be undefined behaviour. Using this switch allows us to ensure that we know what enum is stored. I will add an overload that also takes in `CanMessageId` for the event that we want to reuse the message struct for different messages, but I would like to leave this version for when receiving messages.